app/main.py

```diff
@@ -2,5 +2,13 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = dict()
+    def wrapper(*args, **kwargs) -> int:
+        nonlocal storage
+        if args not in storage:
```

Comment:
Using `args` as a key in the `storage` dictionary can lead to issues if any of the arguments are mutable (e.g., lists). Mutable objects are not hashable and will raise a `TypeError` if used as a dictionary key. Consider converting `args` to a tuple of immutable types or handling mutable arguments differently.